### PR TITLE
Fix for Issue #385

### PR DESCRIPTION
**Pull request by BucketofJava**
## Description

A fix for issue #385, updating time to be recorded in ns/gp/rhs. Also adding benchmark case.py to ./examples.

Related to https://github.com/MFlowCode/MFC/issues/299

### Type of change

Please delete options that are not relevant.

- [x] Something else

### Scope

- [x] This PR comprises a set of related changes with a common goal

If you cannot check the above box, please split your PR into multiple PRs that each have a common goal.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. 
Provide instructions so we can reproduce. 
Please also list any relevant details for your test configuration

- [x] Ran benchmark cases

**Test Configuration**:

* What computers and compilers did you use to test this:
Tested locally on WSL
Lenovo L380 Yoga
Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz 

## Checklist

- [x] This PR does not introduce any repeated code (it follows the [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) principle)
- [x] I cannot think of a way to condense this code and reduce any introduced additional line count

### CODE REVIEW

File: examples/benchmark/case.py
Comment (sbryngelson) [inline examples/benchmark/case.py:1]: Why is this file added?

File: src/simulation/m_start_up.fpp [hunk 1]
```diff
@@ -1127,11 +1127,11 @@ contains
             if (num_procs == 1) then
                 time_final = time_avg
                 io_time_final = io_time_avg
-                print *, "Final Time", time_final
+                print *, "Final Time", time_final*1000000000, " ns/gp/rhs"
             else
                 time_final = maxval(proc_time)
                 io_time_final = maxval(io_proc_time)
-                print *, "Final Time", time_final
+                print *, "Final Time", time_final*1000000000, " ns/gp/rhs"
```
Comment (sbryngelson) [inline src/simulation/m_start_up.fpp]: it looks like this print statement doesn't need to go in the `if` or `else` condition and can just go after

File: examples/benchmark/case.py
Comment (henryleberre) [reply examples/benchmark/case.py:1]: I think this is because of how you worded #385. We could now maybe substitute it with the `./mfc.sh bench` feature. Thoughts?

File: src/simulation/m_start_up.fpp (hunk 1 above)
Comment (henryleberre) [reply src/simulation/m_start_up.fpp]: Additionally, you can rewrite `1000000000` as `1.0d9` (should be more readable).

File: examples/benchmark/case.py
Comment (sbryngelson) [reply examples/benchmark/case.py:1]: @henryleberre I should probably re-read my own issue before critiquing the PR fixing it... I totally forgot about this. 

OK so it does make sense to add the case that is used for "performance expectations" to the benchmark suite I think (or still think, apparently). 

What do you mean by "We could now maybe substitute it with the `./mfc.sh bench` feature"? I assume running this command runs the four (now five?) benchmark cases, but perhaps that's not the case? I haven't recently run it myself actually.

File: src/simulation/m_start_up.fpp (hunk 1 above)
Comment (sbryngelson) [reply src/simulation/m_start_up.fpp]: agree

**Comment by sbryngelson** [comment]
Needs to run `./mfc.sh format` @BucketofJava

File: examples/benchmark/case.py
Comment (henryleberre) [reply examples/benchmark/case.py:1]: I hadn't really considered runtime.. I just took a look at the page on the website and this approach seems very reasonable to me, for the purposes I think this serves. With that in mind, it probably wouldn't be all that useful to provide a case-by-case breakdown for each benchmark configuration. I would probably suggest then that the case be added to the `benchmarks` folder.

Comment (sbryngelson) [reply examples/benchmark/case.py:1]: One can only compare to the expected performance page if you are running very close to the same case, so we should include it in that sense. However, this PR does not do this fully, and the case file needs a proper name.

**Comment by BucketofJava** [comment]
Updated the code according to the comments; For the benchmark case, I moved it to the benchmarks folder, named it "default" as a placeholder and added it to the yml.

**Comment by sbryngelson** [comment]
You are failing the `Pretty` CI because you didn't follow my comment above (or in the PR template). How long does it take to run this benchmark case? On CPU? GPU?

**Comment by sbryngelson** [comment]
This PR doesn't actually compute the other things I mention in the issue:
```

Update: Probably want to switch both the code and the .md file in the docs to be ns per grid-point per equation per rhs evaluation (I updated the docs to have this already but without the "per-equation" part).
```

**Comment by BucketofJava** [comment]
Sorry, I did not see the per-equation part before. I have run ./mfc.sh format with every commit, however I did not run it with -j ${nproc} as in the test case, this commit has done this, so hopefully the issue should be fixed.

**Comment by sbryngelson** [comment]
@BucketofJava if you change the docs markdown page to have _per equation_ then you also need to change all of the performance numbers in that table (dividing them by the number of equations). I think this means dividing them all by 8?

**Comment by sbryngelson** [comment]
PR merges are on hold until benchmark CI is working (issue #419)

**Comment by sbryngelson** [comment]
@BucketofJava or someone... can you force sync this fork with MFC root master so I can merge?

**Comment by sbryngelson** [comment]
@AiredaleDev can you assist with finishing off this PR?

**Comment by AiredaleDev** [comment]
Sure thing -- it merged cleanly with master with no issue. I'd like to push the changes to this PR but I need access to @BucketofJava 's fork, so I made another PR instead: #432

**Comment by sbryngelson** [comment]
Superseded by #411
